TestMetadataChanged fails on systems with UID 51234 taken by a user

Output of `restic version`
--------------------------

$ ./restic version
restic 0.9.5-dev (compiled manually) compiled with go1.12.5 on linux/amd64

How did you run restic exactly?
-------------------------------

`go test ./...` in a fresh git checkout

What backend/server/service did you use to store the repository?
----------------------------------------------------------------

n/a

Expected behavior
-----------------

tests pass

Actual behavior
---------------

```
--- FAIL: TestMetadataChanged (0.01s)
    testing.go:30: using low-security KDF parameters for test
    config.go:65: disabling check of the chunker polynomial
    archiver_test.go:2043: chdir to /tmp/restic-test-488507700
    archiver_test.go:2091: metadata does not match:
        {*restic.Node}.User:
        	-: ""
        	+: "<username>"
    panic.go:406: chdir back to /<...>/restic/internal/archiver
```

Steps to reproduce the behavior
-------------------------------

Run tests on a system where UID 51234 is defined and occupied by a user.

$ id 51234 > /dev/null && echo "UID exists"
UID exists

Do you have any idea what may have caused this?
-----------------------------------------------

Yes, this part of the test:

```
	if runtime.GOOS != "windows" {
		want.UID = 51234
		want.GID = 51235
	}
	// no user and group name
	want.User = ""
	want.Group = ""

	// make another snapshot
	snapshotID, node3 := snapshot(t, repo, fs, snapshotID, "testfile")

	// make sure that metadata was recorded successfully
	if !cmp.Equal(want, node3) {
		t.Fatalf("metadata does not match:\n%v", cmp.Diff(want, node3))
	}
```


Do you have an idea how to solve the issue?
-------------------------------------------

One of:
- don't check for username at all (check for UID)
- pick UID which is guaranteed to be unoccupied (negative? will this work?)

Did restic help you or made you happy in any way?
-------------------------------------------------

yup :)

--- a/internal/archiver/archiver_test.go
+++ b/internal/archiver/archiver_test.go
@@ -2085,6 +2085,10 @@ func TestMetadataChanged(t *testing.T) {
 
 	// make another snapshot
 	snapshotID, node3 := snapshot(t, repo, fs, snapshotID, "testfile")
+	// Override username and group to empty string - in case underlying system has user with UID 51234
+	// See https://github.com/restic/restic/issues/2372
+	node3.User = ""
+	node3.Group = ""
 
 	// make sure that metadata was recorded successfully
 	if !cmp.Equal(want, node3) {

